packages/clay-drop-down/src/Item.tsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import ClayCheckbox from '@clayui/checkbox';
+import ClayIcon from '@clayui/icon';
+import ClayRadioGroup from '@clayui/radio-group';
+import Divider from './Divider';
+
+interface BaseDropDownItem {
+	active?: boolean;
+	disabled?: boolean;
+	href?: string;
+	items?: (BaseDropDownItem | FormItem)[];
+	label?: string;
+	onClick?: () => void;
+	type?: 'group' | 'radio' | 'checkbox' | 'divider';
+	rightSymbol?: string;
+	leftSymbol?: string;
+}
+
+interface FormItem extends BaseDropDownItem {
+	label: string;
+	name: string;
+	items: FormItem[];
+	onChange: (val?: React.ChangeEvent | React.ReactText) => void;
+	type: 'radio' | 'checkbox';
+	value: string | number;
+}
+
+export type DropDownItemType = FormItem | BaseDropDownItem;
+
+interface Props {
+	data: DropDownItemType;
+	spritemap?: string;
+}
+
+const DropDownItem: React.FunctionComponent<Props> = ({data, spritemap}) => {
```

Comment:
I was a little curious why you did not follow with the idea of ​​composition (renders props...). My concern I had with our Metal component was that it was getting complex as more use cases appeared, some people wanted to have more control over dropdown items.

One suggestion I have about using the composition, I think we can give off some of the Lexicon and offer escape doors to "get away". This is just a suggestion and a sketch:

#### `<Dropdown />`
Just the basic component that will control the opening and closing of Dropdown, click away, ESC... and maybe listen to the items click events directly and onChange for checkbox and radio cases...

```jsx
<Dropdown
  onItemClick={...} // maybe
>
  {children}
</Dropdown>
```

#### `<Dropdown.Search />`, `<Dropdown.Caption />`, `<Dropdown.Action />`

```jsx
<Dropdown>
  <Dropdown.Search ... />
  <Dropdown.Menu>...</Dropdown.Menu>
  <Dropdown.Caption ... />
  <Dropdown.Action>
    <ClayButton ... />
  </Dropdown.Action>
</Dropdown>
```

#### `<Dropdown.Menu />`
Just to wrap the components of items.

```jsx
<Dropdown.Menu>
  <Dropdown.Item>Item</Dropdown.Item>
</Dropdown.Menu>
```

#### `<Dropdown.Item />`
These are our standard variations, checkbox, radio, default... Nothing new, we can keep our APIs here.

```jsx
<Dropdown.Item>Item</Dropdown.Item>
<Dropdown.Divider />
<Dropdown.Item variant="checkbox" value="checkbox">Checkbox</Dropdown.Item>
```

#### `<Dropdown.Group />`
Just to create group of items, not to need a props of `items` inside the `<Dropdown.Item />` element.

```jsx
<Dropdown.Group label="Group 1">
  <Dropdown.Item>Item</Dropdown.Item>
</Dropdown.Group>
```

What do you think?

Reply:
I think this is a great idea. I originally set out to just copy the soy version, but I think this makes more sense and lets us manage less logic on our end. Ill go ahead and update and see how it works.